src/tests/errors.rs

```diff
@@ -51,3 +51,36 @@ fn function_with_failable_typed_arg() {
 
     test_amber(code, "Hello, World!");
 }
+
+#[test]
+#[should_panic(expected = "ERROR: Index accessor must be a number or range")]
+fn get_array_index_by_string() {
+    let code = r#"
+        let array = [0, 1, 2, 3]
+        let slice = array["foo"]
+    "#;
+
+    test_amber(code, "");
+}
+
+#[test]
+#[should_panic(expected = "ERROR: Index accessor must be a number")]
+fn set_array_index_by_string() {
+    let code = r#"
+        let array = [0, 1, 2, 3]
+        array["foo"] = [11, 22]
+    "#;
+
+    test_amber(code, "");
+}
+
+#[test]
+#[should_panic(expected = "ERROR: Index accessor must be a number")]
```

Comment:
I'd like it if the error message specifically said that it is not possible to use range for indexing in an assignment.

Reply:
Changed to:
```
Index accessor must be a number or range for retrieval
Index accessor must be a number (and not a range) for assignment
```